docs/source/developer/inlining.rst

```diff
@@ -76,10 +76,41 @@ The ``inline`` keyword argument can be one of three values:
   In all cases the function should return True to inline and return False to not
   inline, this essentially permitting custom inlining rules (typical use might
   be cost models).
+* Recursive functions with ``inline='always'`` will result in a non-terminating
+  compilation. If you wish to avoid this, use a cost model.
```

Comment:
```suggestion
  compilation. If you wish to avoid this, supply a function to limit the recursion depth (see below).
```

Cost models are just an example of one use, this really is just a function to determine whether inline or not based on anything the user finds useful, see the example code http://numba.pydata.org/numba-doc/dev/developer/inlining.html#example-using-numba-jit, where there's no cost being measured, it's just a scan for an IR token to determine whether to inline.